pymesync.py

```diff
@@ -38,6 +44,41 @@ def __init__(self, baseurl, user, password, auth_type):
             "activity": [],
         }
 
+    def authenticate(self, username, password, auth_type):
```

Comment:
I would give your kwargs default values, or when you assign them to self variables do it in a try/except block to avoid problems with that. Even a default value of None is better than the 'function takes exactly N arguments' errors.